### ISSUE TITLE
Add support for Ruby 3

### ISSUE THREAD

**Issue by ehannes** [accodeing/fortnox-api#212]
Travis build is not even working now:

```
Setting environment variables from .travis.yml
$ export CC_TEST_REPORTER_ID=[filtered-id]
1.10s$ rvm get head
Downloading https://get.rvm.io
Downloading https://bitbucket.org/mpapis/rvm/raw/master/binscripts/rvm-installer.asc
Micros.asc
The command "rvm get head" failed and exited with 3 during .
Your build has been stopped.
```

**Comment by ehannes** [comment]
e93fb785108497dede6248928105b64f9ea2c4ef fixed the above error. Simply added an explicit Ubuntu version to run Travis with. It was by default running on Xenial before, which is Ubuntu 16.04! That's 5 years old now, time to upgrade! Interesting that Travis uses that distro by default...

Now we have another problem, in both 3.0.4 and 3.1.2:
```
Failure/Error: require 'dry-struct'
ArgumentError:
  wrong number of arguments (given 2, expected 1)
```

**Comment by ehannes** [comment]
The above problem might be because the current used version of `dry-struct` is not supporting Ruby 3. That version is released 2017-11-04, more than 4 years ago and long before Ruby 3 was around!

**Comment by ehannes** [comment]
Resolved in #228
